accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/DatabaseService.java

```diff
@@ -59,12 +59,15 @@
             AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException;
 
     /**
-     * Finds last version of all valid accessions with their possible data model representations.
+     * Finds last version of provided accession with its possible data model representations.
      *
-     * @param accessions valid accession id
-     * @return All valid accessions. No deprecated or merged ids will be returned.
+     * @param accession valid accession id
+     * @return valid accession. No deprecated ids will be returned.
+     * @throws AccessionDoesNotExistException when accession does not exist.
+     * @throws AccessionDeprecatedException   accession is no longer active.
      */
-    List<AccessionWrapper<MODEL, HASH, ACCESSION>> findAllByAccession(List<ACCESSION> accessions);
+    AccessionWrapper<MODEL, HASH, ACCESSION> findLastVersionByAccession(ACCESSION accession)
+            throws AccessionDoesNotExistException, AccessionDeprecatedException;
```

Comment:
Shouldn't this also throw `AccessionMergedException`? Given that we **definitely** want to catch these exceptions when they occur, maybe we should make them checked.

Reply:
Those are checked exceptions already.  as u said as we definitely wanted catch them , we didn't throw the exception.

Reply:
made the method to throw AccessionMergedException and adding a redirect in the controller advice.